vllm/core/scheduler.py

```diff
@@ -1027,6 +1036,10 @@ def free_seq(self, seq: Sequence) -> None:
         self.block_manager.free(seq)
 
     def free_finished_seq_groups(self) -> None:
+        self.finished_requests_ids += [
+            seq_group.request_id for seq_group in self.running
+            if seq_group.is_finished()
+        ]
```

Comment:
There's a leak here if a sequence is running then is preempted, and aborted by the user while it is preempted. We should check all queues.

Reply:
I changed it to check through all of the queues, not just the running queue.